### PR TITLE
Add additional argument string to the executable

### LINKED ISSUE #190: How to pass command line arguments
This one is closely related to [this issue](https://github.com/pyiron/pyiron_atomistics/issues/75). Now I'd like to pass command line arguments to the executable. Currently, the LAMMPS executable contains:

```
#!/bin/bash
module load pyiron/dev
mpiexec -n $1 lmp_mpi -in control.inp
```

But I'd like to extend it to something like

```
#!/bin/bash
module load pyiron/dev
mpiexec -n $1 lmp_mpi -in control.inp ${@:2}
```

and from LammpsBase I'd love to pass custom arguments. Is there a possibility to do something like this?

### PR DESCRIPTION
**Pull request by niklassiemer**
Closes #190

I introduce a `additional_arguments` string at the `Executable` class which is added to the system call of the executable. I store the additional command line arguments in `hdf["RUNARGS"]`.
@samwaseda with these changes it should be possible to extend the shell scripts as you proposed in #190.

### CODE REVIEW

**Comment by niklassiemer** [comment]
Well, on windows this does not work, yet. I will look into it tomorrow.

File: pyiron_base/job/generic.py [hunk 1]
```diff
@@ -1514,6 +1516,10 @@ def _type_from_hdf(self):
                 self.executable.version = self._hdf5["VERSION"]
             except ValueError:
                 self.executable.executable_path = self._hdf5["VERSION"]
+            try:
+                self.executable.additional_arguments = self._hdf5["RUNARGS"]
```
Comment (jan-janssen) [inline pyiron_base/job/generic.py:1524]: rather than `try except` I would prefer `if "RUNARGS" in self._hdf5.list_nodes():`

**Comment by niklassiemer** [comment]
Although all tests pass, I do not yet trust this implementation. I am also confused the Linux tests did not fail on the previous commit...

**Comment by niklassiemer** [comment]
@jan-janssen How do I get a echo executable to work for linux and windows? Or do I actually have to have two of them?...

**Comment by pmrv** [comment]
> @jan-janssen How do I get a echo executable to work for linux and windows? Or do I actually have to have two of them?...

Try 
```python
job.executable = "echo $1 $2 $@"
```
for linux and a comparable string for windows, you can decide the path based on `sys.platform`.  I've never used `subprocess` on windows though, does it run stuff in cmd.exe? :S

**Comment by niklassiemer** [comment]
I actually tried to write the stuff to a file and call that (works on Linux) however I will try this solution :)

**Comment by niklassiemer** [comment]
Great, I got it working locally on Windows and Linux with a file based solution. However, CI does not seem to like that.

**Comment by niklassiemer** [comment]
The most important test I don't get running on CI so far. Any ideas how to verify that the first two arguments are the number of cores and threads for mpi executables?

**Pull request closed without merging**